test/Lint.lean

```diff
@@ -87,6 +87,15 @@ note: this linter can be disabled with `set_option linter.cdot false`
 set_option linter.cdot true in
 example : Add Nat where add := (. + ·)
 
+set_option linter.dollarSyntax false in
+/--
+warning: Please use '<|' and not '$' for the pipe operator.
+note: this linter can be disabled with `set_option linter.dollarSyntax false`
+-/
+#guard_msgs in
+set_option linter.dollarSyntax true in
+example (f g : Nat → Nat) : Nat := f $ g 0
```

Comment:
Thanks for including a test! Can you extend it slightly, to
- include more than one dollar (writing "f $ g $ 0" suffices perfectly here)
- also include other kinds of syntax nodes? (Some ideas: include a theorem, definition or lemma; include this syntax inside a have, let or nested goal, and nested inside a more complex term proof. Ping me if this is unclear!)

Reply:
I've done this now.